qiime/core/type/signature.py

```diff
@@ -94,10 +110,30 @@ def defaults(self):
             (name, spec.default) for name, spec in self.parameters.items()
             if spec.has_default()])
 
-    def _parse_signature(self, callable, inputs, parameters, outputs):
+    @property
+    def descriptions(self):
+        params = {**self.inputs, **self.parameters, **self.outputs}
```

Comment:
Cool, @jakereps if it's useful to q2cli let's keep the method. It should probably be updated to either 1) maintain order and return an OrderedDict, or 2) return a regular dict